### ISSUE TITLE
LibC: vfscanf() consumes a whole line of input on each call

### ISSUE THREAD

**Issue by awesomekling** [SerenityOS/serenity#5633]
This doesn't look right:

```
int vfscanf(FILE* stream, const char* fmt, va_list ap)
{
    char buffer[BUFSIZ];
    if (!fgets(buffer, sizeof(buffer) - 1, stream))
        return -1;
    return vsscanf(buffer, fmt, ap);
}
```

**Comment by ADKaster** [comment]
> An item shall be read from the input, unless the conversion specification includes an n conversion specifier. An input item shall be defined as the longest sequence of input bytes (up to any specified maximum field width, which may be measured in characters or bytes dependent on the conversion specifier) which is an initial subsequence of a matching sequence. _**The first byte, if any, after the input item shall remain unread**_. If the length of the input item is 0, the execution of the conversion specification shall fail; this condition is a matching failure, unless end-of-file, an encoding error, or a read error prevented input from the stream, in which case it is an input failure.

> If conversion terminates on a conflicting input, _**the offending input is left unread in the input.**_ Any trailing white space (including <newline> characters) shall be left unread unless matched by a conversion specification. The success of literal matches and suppressed assignments is only directly determinable via the %n conversion specification.

Yep, absolutely against spec to read the whole line of size BUFSIZ and just forward to sscanf. fscanf and friends are supposed to carefully fgetc and take action based on the format specifiers in the format string.

https://pubs.opengroup.org/onlinepubs/9699919799/functions/fscanf.html
